### PR TITLE
Fix: disable want-to-read button until JS is loaded

### LINKED ISSUE #12042: Clicking "want to read" before JS has loaded and run takes you to a .json dead end
### Problem

The "want to read" button on work pages needs JS to run before it works correctly.

Before that, if you click on it quiclky while the JS hasn't yet fully loaded, it takes you to a .json page that just displays:

```
{
      "bookshelves_affected": null
}
```

I think adding `redir=true` to the hidden elements of the form makes it work without JS, but when the JS loads (or when it fires when the button gets clicked) it would need to remove this extra form element or it breaks the ajax json call.

### Reproducing the bug

1. Go to any page with a "want to read" button
2. Click it as soon as you see it
3. If the site is loading too fast to see the issue, you can fake this by disabling javascript in the browser dev tools

* Expected behavior: The book would be (un)marked as "want to read" and the site continues to display
* Actual behavior: You see a .json formatted page never intended to be seen by the user

### Context

- Browser (Chrome, Safari, Firefox, etc): any
- OS (Windows, Mac, etc): any
- Logged in (Y/N): any
- Environment (prod, dev, local): prod

### Breakdown

<details>
<summary><strong>Implementation Details</strong> (for maintainers)</summary>

#### Requirements Checklist
* [ ]

#### Related files

*

#### Stakeholders

*
<hr>

#### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

</details>

### PR DESCRIPTION
**Pull request by mekarpeles**
## Summary

Prevent the ".json dead end" issue by disabling the "Want to Read" button in HTML until JavaScript has loaded and initialized its event handlers.

## Problem

When a user clicks the "Want to Read" button before JavaScript has loaded, the form submits directly to a `.json` endpoint, displaying raw JSON (`{"bookshelves_affected": null}`) to the user instead of the expected page behavior.

## Solution

1. **HTML**: Added `disabled` attribute to the primary action button
2. **JavaScript**: Enable the button in `ReadingLogForms.initialize()` after event handlers are set up
3. **CSS**: Override disabled appearance so the button looks unchanged to users

This approach was suggested by @mekarpeles in the issue discussion - keeping it simple by preventing clicks until JS is ready.

## Testing

- Python lint: ✅ Passed
- JS/CSS lint: ✅ Passed  
- Python tests: ✅ 2883 passed
- JS tests: ✅ 302 passed

## Considerations

- The button visually appears the same but won't respond to clicks until JS loads
- Users may need to click again after the page fully loads (rare edge case)
- No changes to the API or backend logic

## Related Issues

Fixes #12042

### CODE REVIEW

**Comment by mekarpeles** [comment]
looks like a rebase went terribly wrong, I can try to readdress, sorry!

File: openlibrary/templates/my_books/primary_action.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
     $if user_key:
       <input type="hidden" name="user-key" value="$(user_key)" />
 
-    <button class="book-progress-btn primary-action $(activated_status)" type="submit">
+    <button class="book-progress-btn primary-action $(activated_status)" type="submit" disabled>
```
[the remark this answers is not in the document]
Comment (mekarpeles) [reply openlibrary/templates/my_books/primary_action.html:27]: @copilot is this true?

Only the first `<input type="hidden" name="user-key" value="$(user_key)" />` is nested under `if user_key`

Presumably because `<button class="book-progress-btn primary-action $(activated_status)" type="submit" $if user_key:disabled>` is not nested under the guard it should show up irrespective of user_key?
